src/com/haskforce/codeInsight/HaskellCompletionContributor.java

```diff
@@ -71,17 +72,32 @@ public void addCompletions(@NotNull CompletionParameters parameters,
                         // whether or not we were in the appropriate context.  This is useful to determine if following
                         // completions should be added.
                         completeKeywordImport(position, result);
+                        completeHaskellKeywords(position, result);
                         completeKeywordQualified(position, result);
                         if (completePragma(position, cacheHolder, result)) return;
                         if (completeModuleImport(position, cacheHolder, result)) return;
                         if (completeQualifiedNames(position, imports, cacheHolder, result)) return;
                         if (completeNameImport(position, cacheHolder, result)) return;
                         completeLocalNames(position, imports, cacheHolder, result);
+                        completeFunctionLocalNames(position,result);
                     }
                 }
         );
     }
 
+    /**
+     * Will only propose the haskell keywords when inside a haskell body. Will not take into account the specific position
+     * yet, just make sure it's inside the body, so as to not propose keywords other than import in imports
+     */
+    public static void completeHaskellKeywords(@NotNull final PsiElement position, @NotNull final CompletionResultSet result) {
+        if (HaskellUtil.isInsideBody(position)){
+            Set<String> haskellKeywords = HaskellNamesValidator.HASKELL_KEYWORDS;
+            for (String haskellKeyword : haskellKeywords) {
+                result.addElement(LookupElementBuilder.create(haskellKeyword));
+            }
+        }
+    }
+
```

Comment:
This is a little concerning for me, particularly because it can be quite annoying to have keywords complete in places that don't make sense.  If there are specific keywords you are considering here (e.g. `do`) I'd say provide completions specifically for those.  Many may fall within the same logic.  For instance, `do` should only occur within expressions.